File: website/src/main/java/org/bag/dto/assemblers/FrontendProjectAssembler.java
package org.bag.dto.assemblers;

import org.bag.domain.Project;
import org.bag.domain.Image;
import org.bag.dto.FrontendProject;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

/**
 * Maakt een FrontendProject aan obv een Project
 */
public class FrontendProjectAssembler {
    private Project project;

    public FrontendProjectAssembler(Project project) {
        this.project = project;
    }

    public FrontendProject assemble(Locale locale) throws Exception {
        String langCode = locale.getLanguage();
        return new FrontendProject(project.id(),
                project.getTitle(langCode),
                project.getLocatie(langCode),
                project.getText(langCode),
                project.bannerImage().url(),
                project.frontPageImage().url(),
                getImageList(project));
    }

    private List<String> getImageList(Project project) {
        ArrayList<String> images = new ArrayList<>();
        for (Image image : project.images()) {
                images.add(image.url());
        }
        return images;
    }


}

File: domain/src/main/kotlin/org/bag/dto/assemblers/FrontendProjectAssembler.kt
package org.bag.dto.assemblers

import org.bag.domain.Project
import org.bag.dto.FrontendProject

import java.util.ArrayList
import java.util.Locale

/**
 * Maakt een [FrontendProject] aan obv een [project]
 */
class FrontendProjectAssembler(private val project: Project) {

    fun assemble(locale: Locale): FrontendProject {
        val langCode = locale.language
        return FrontendProject(project.id,
                project.getTitle(langCode),
                project.getLocatie(langCode),
                project.getText(langCode),
                project.getBannerImage().url,
                project.getFrontendImage().url,
                getImageList(project))
    }

    private fun getImageList(project: Project): List<String> {
        val images = ArrayList<String>()
        for (image in project.images) {
            images.add(image.url)
        }
        return images
    }


}
